x-pack/plugins/graph/public/helpers/use_workspace_loader.ts

```diff
@@ -91,22 +90,21 @@ export const useWorkspaceLoader = ({
           .then((response) => {
             perPage = response.perPage;
             total = response.total;
-            page = response.page;
             savedObjects = response.savedObjects;
           });
 
         yield savedObjects;
 
         if (total > page * perPage) {
-          yield* makeRequest();
+          yield* makeRequest(page++);
```

Comment:
Didn't test I think the request is still always for page 1. Because the `page++` first returns the original value, and then increments the value. Would not be the case when `++` is prepended 
```suggestion
          yield* makeRequest(++page);
```